gips/data/modis/modis.py

```diff
@@ -353,22 +376,29 @@ def process(self, *args, **kwargs):
 
             # LAND VEGETATION INDICES PRODUCT
             if val[0] == "indices":
-
                 VERSION = "2.0"
                 meta['VERSION'] = VERSION
                 sensor = 'MCD'
                 fname = '%s_%s_%s' % (bname, sensor, key)
-
                 refl = gippy.GeoImage(allsds)
-
                 missing = 32767
 
-                redimg = refl[7].Read()
-                nirimg = refl[8].Read()
-                bluimg = refl[9].Read()
-                grnimg = refl[10].Read()
-                mirimg = refl[11].Read()
-                swrimg = refl[12].Read() # formerly swir2
+                if versions[asset] == 6:                
+                    redimg = refl[7].Read()
+                    nirimg = refl[8].Read()
+                    bluimg = refl[9].Read()
+                    grnimg = refl[10].Read()
+                    mirimg = refl[11].Read()
+                    swrimg = refl[12].Read() # formerly swir2
+                elif versions[asset] == 5:
+                    redimg = refl[0].Read()
+                    nirimg = refl[1].Read()
+                    bluimg = refl[2].Read()
+                    grnimg = refl[3].Read()
+                    mirimg = refl[4].Read()
+                    swrimg = refl[5].Read() # formerly swir2
```

Comment:
I tried both ways and went for clarity but I’m OK either ways.

> On Jul 5, 2016, at 10:16 AM, ags-tolson <notifications@github.com> wrote:
> 
> In gips/data/modis/modis.py <https://github.com/Applied-GeoSolutions/gips/pull/53#discussion_r69570155>:
> 
> > -                mirimg = refl[11].Read()
> > -                swrimg = refl[12].Read() # formerly swir2
> > +                if versions[asset] == 6:                
> > +                    redimg = refl[7].Read()
> > +                    nirimg = refl[8].Read()
> > +                    bluimg = refl[9].Read()
> > +                    grnimg = refl[10].Read()
> > +                    mirimg = refl[11].Read()
> > +                    swrimg = refl[12].Read() # formerly swir2
> > +                elif versions[asset] == 5:
> > +                    redimg = refl[0].Read()
> > +                    nirimg = refl[1].Read()
> > +                    bluimg = refl[2].Read()
> > +                    grnimg = refl[3].Read()
> > +                    mirimg = refl[4].Read()
> > +                    swrimg = refl[5].Read() # formerly swir2
> There's a way to cut down on the cutpasting, with something like this:
> 
> offset = {5: 0, 6: 7}[versions[asset]]
> redimg, nirimg, bluimg, grnimg, mirimg, swrimg = [refl[offset + i] for i in range(6)]
> What you wrote is super clear and easy to understand however, so I can see your way being more maintainable.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/Applied-GeoSolutions/gips/pull/53/files/3a9deaff37b928cc1fba52d1cdb2083f4a8127db#r69570155>, or mute the thread <https://github.com/notifications/unsubscribe/ADD4XADLA_9fv7EQ4hF1hSh6Myi1SNVuks5qSmcngaJpZM4JDj03>.
>